### ISSUE TITLE
[1.10-rc1] dynamic daemon configuration does not work with overlay driver

### ISSUE THREAD

**Issue by ibuildthecloud** [docker/docker#19692]
If I start docker with no cluster store and then add 

```
{
    "cluster-advertise": "blah:2377",
    "cluster-store": "etcd://localhost:2379"
}
```

Do a HUP and then the below fails

```
docker network create -d overlay test
Error response from daemon: failed to parse pool request for address space "GlobalDefault" pool "" subpool "": cannot find address space GlobalDefault (most likely the backing datastore is not configured)
```

**Comment by HackToday** [comment]
hi @ibuildthecloud  just want to know if docker support dynamic configuration loading ?
which means I edit systemed related drop-in files, The docker started by systemctl before could pick that changes ? (no need to restart service)  If it is such case, do we have a link for such doc?

**Comment by cpuguy83** [comment]
@HackToday https://github.com/docker/docker/blob/master/docs/reference/commandline/daemon.md#configuration-reloading

**Comment by thaJeztah** [comment]
ping @calavera, assigned this one to you

**Comment by tiborvass** [comment]
I don't think we can realistically have this for 1.10, however it would be great to have an error message.

**Comment by thaJeztah** [comment]
@tiborvass agreed; should be fixed one way or the other I think

**Comment by ibuildthecloud** [comment]
@tiborvass Whoa, stop the presses, why can't we have this one?  If cluster store can not be dynamically changed then that largely defeats the purpose of the dynamic configuration, basically only labels and debug are left after cluster store.

**Comment by mavenugo** [comment]
@ibuildthecloud @tiborvass libnetwork doesn't support dynamic cluster-store configuration. We discussed about this during one of the maintainers meeting as well. It is targeted for 1.11 in libnetwork.

**Comment by thaJeztah** [comment]
@mavenugo I'll move this issue to the 1.11 milestone, and added a new issue for adding an error message; https://github.com/docker/docker/issues/19732

**Comment by calavera** [comment]
This will be fixed in 1.11. #20419 was the PR that fixed the issue. Closing.
